### PR TITLE
Fix doc generation

### PR DESCRIPTION
**Pull request by phiggins**
When looking into an unrelated problem with an expeditor update script and looked at the logs for this project's and noticed some problems: https://logs.cd.chef.co/raw/chef-cookstyle-master-pull_request_merged-chef-cookstyle-master-710-1105ea80-d774-11ea-9c0a-0ce4d00b69f0-0260a04d0d3eebe0

The main thing was docs generation was failing:

```
rake aborted!
NoMethodError: undefined method `support_autocorrect?' for #<RuboCop::Cop::Chef::ChefCorrectness::MacosUserdefaultsInvalidType:0x000056237bd149b8>
tasks/cops_documentation.rake:49:in `properties'
tasks/cops_documentation.rake:29:in `cops_body'
tasks/cops_documentation.rake:187:in `print_cop_with_doc'
tasks/cops_documentation.rake:166:in `block in print_cops_of_department'
tasks/cops_documentation.rake:165:in `each'
tasks/cops_documentation.rake:165:in `print_cops_of_department'
tasks/cops_documentation.rake:238:in `block in main'
tasks/cops_documentation.rake:237:in `each'
tasks/cops_documentation.rake:237:in `main'
tasks/cops_documentation.rake:246:in `block in <top (required)>'
/opt/asdf/installs/ruby/2.6.6/bin/bundle:23:in `load'
/opt/asdf/installs/ruby/2.6.6/bin/bundle:23:in `<main>'
Tasks: TOP => generate_cops_documentation
(See full trace by running task with --trace)
```

### CODE REVIEW

File: lib/rubocop/cop/chef/deprecation/resource_uses_only_resource_name.rb [hunk 1]
```diff
@@ -55,7 +55,7 @@ def cookbook_name
 
           # given a resource name make sure there's a provides that matches that name
           #
-          # @returns [TrueClass, FalseClass]
+          # @return [TrueClass, FalseClass]
           def valid_provides?(resource_name)
             provides_ast = provides(processed_source.ast)
             return false unless provides_ast
```
Comment (phiggins) [inline lib/rubocop/cop/chef/deprecation/resource_uses_only_resource_name.rb:61]: Fixes these warnings:

```
[warn]: Unknown tag @returns in file `lib/rubocop/cop/chef/deprecation/resource_uses_only_resource_name.rb` near line 45
[warn]: Unknown tag @returns in file `lib/rubocop/cop/chef/deprecation/resource_uses_only_resource_name.rb` near line 59
```

File: cookstyle.gemspec [hunk 2]
```diff
@@ -11,6 +11,7 @@ Gem::Specification.new do |spec|
   spec.email         = ['thom@chef.io', 'tsmith@chef.io']
   spec.summary       = 'Cookstyle is a code linting tool that helps you to write better Chef Infra cookbooks by detecting and automatically correcting style, syntax, and logic mistakes in your code.'
   spec.license       = 'Apache-2.0'
+  spec.homepage      = 'https://docs.chef.io/workstation/cookstyle/'
```
Comment (phiggins) [inline cookstyle.gemspec:14]: Fixes this warning:

```
[2020-08-05 23:34:39 +0000] action built_in:build_gem started
WARNING:  no homepage specified
WARNING:  See http://guides.rubygems.org/specification-reference/ for help
  Successfully built RubyGem
  Name: cookstyle
  Version: 6.15.1
  File: cookstyle-6.15.1.gem
```

I figured if someone wants the gitbub repo that's easy enough to find.

File: docs/cops_chefdeprecations.md [hunk 3]
```diff
@@ -35,7 +35,7 @@ Exclude | `**/metadata.rb`, `**/Berksfile` | Array
 
 ### References
 
-* [https://rubystyle.guide#chefdkgenerators](https://rubystyle.guide#chefdkgenerators)
+* [https://rubystyle.guide#chefdeprecationschefdkgenerators](https://rubystyle.guide#chefdeprecationschefdkgenerators)
```
Comment (tas50) [inline docs/cops_chefdeprecations.md:38]: I need to figure out why these aren't working. It should get the style guide from the department.

**Comment by tas50** [comment]
Nice find. Thanks

**Pull request merged**